File: backend/app/schemas/category.py
from pydantic import BaseModel
from .base import BaseSchema

class CategoryBase(BaseSchema):
    name: str

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(BaseModel):
    name: str

class CategoryOut(CategoryBase):
    id: int

class UpdateFeedCategory(BaseSchema):
    category_id: int
    feed_id: int

class UpdateChannelCategory(BaseSchema):
    category_id: int
    channel_id: str
    

File: frontend/types/category.ts
export interface CategoryCreate {
    name: string
  }
  
  export interface CategoryOut {
    id: number
    name: string
  }
  
  export interface UpdateFeedCategory {
    category_id: number
    feed_id: number
  }
  
  export interface UpdateChannelCategory {
    category_id: number
    channel_id: string
  }
  
  